utils.c

```diff
@@ -758,13 +758,16 @@ send_pid_on_socket (int sockfd)
   struct msghdr msg = {};
   struct iovec iov = { buf, sizeof (buf) };
   const ssize_t control_len_snd = CMSG_SPACE(sizeof(struct ucred));
-  char control_buf_snd[control_len_snd];
+  union {
+      char buf[control_len_snd];
+      struct cmsghdr align;
+  } control_buf_snd;
```

Comment:
any reason to not use `__aligned__` here? Ofc this version is also valid, but I think the attribute is a tad clearer if you're not familiar with how unions make this work. (the attribute should work going back to gcc 4.x, at least.)

Reply:
Unions are what `cmsg(3)` suggests, and are the historical Standard C way to do this.

If we're going to use something other than unions, I think C11 `_Alignas` would be the way to do it (works in gcc 4.8 but not 4.6).

Reply:
Switched to `_Alignas`.

Reply:
oh geez C11 is already over a decade old, it still feels too new to me